### PR TITLE
[JENKINS-59572] Migrate Docs from Wiki to GitHub

### PR DESCRIPTION
**Pull request by eshack94**
**Note to reviewer:** In order for this to count towards Hacktoberfest contributions, requesting that the label `hacktoberfest-accepted` be applied post-review. Please and thank you! 🙏🏼

The purpose of these changes is to migrate the job restrictions plugin
docs from the legacy Jenkins Wiki site to GitHub.

See also:
- [JENKINS-59572](https://issues.jenkins-ci.org/browse/JENKINS-59572)
- [INFRA-2328](https://issues.jenkins-ci.org/browse/INFRA-2328)
- [Google Groups Thread](https://groups.google.com/u/1/g/jenkinsci-dev/c/VSdfVMDIW-A)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,201 @@
...
 =======================
-See [Job Restrictions Plugin Page][2] for more information.
+
+| Plugin Information |
+|---------------------------------------------------------------------------------------------------------------|
+| View Job Restrictions [on the plugin site](https://plugins.jenkins.io/job-restrictions) for more information. |
+
+**This plugin is up for adoption.** Want to help improve this plugin?
+[Click here to learn
+more](https://wiki.jenkins.io/display/JENKINS/Adopt+a+Plugin "Adopt a Plugin")!
```
Comment (halkeye) [inline README.md:10]:
```suggestion
```

This is handled by github labels now, so shouldn't be in the readme.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,201 @@
 Job Restrictions Plugin for [Jenkins CI][1]
 =======================
-See [Job Restrictions Plugin Page][2] for more information.
+
+| Plugin Information |
+|---------------------------------------------------------------------------------------------------------------|
+| View Job Restrictions [on the plugin site](https://plugins.jenkins.io/job-restrictions) for more information. |
```
Comment (halkeye) [inline README.md:6]:
```suggestion
```

This is natively handled in https://plugins.jenkins.io/job-restrictions so unncessary now.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,201 @@
...
+# Usage
+
+## Job restrictions for nodes
+
+These restrictions can be configured in node's configuration page.
+
+-   Node won't accept jobs, which violate the specified restrictions
+-   Other requirements (executors, labels, etc.) will be checked as well
+
+![](https://wiki.jenkins.io/download/attachments/69273423/Jenkins_JobRestrictions_NodeRestriction.png?version=1&modificationDate=1378665445000&api=v2)
```
Comment (halkeye) [inline README.md:47]: Assuming you used the wiki exporter, can you use the zip format so it gives you the attachments as well? this way the docs won't still refer to the wiki

File: README.md (hunk 2 above)
Comment (eshack94) [reply README.md]: Thanks for the feedback! Removed this section.

File: README.md (hunk 1 above)
Comment (eshack94) [reply README.md]: Thanks for the heads-up. I removed this section!

File: README.md (hunk 3 above)
Comment (eshack94) [reply README.md:39]: @halkeye Pardon my ignorance, but what is the best practice for linking from a Zip file in Github-flavored Markdown? Thanks again for the feedback.

Comment (halkeye) [reply README.md:39]: use https://jenkins-wiki-exporter.jenkins.io/ to export a zip
then extract the zip into a directory, and commit that.
@MarkEWaite I think has better tutorials

Comment (MarkEWaite) [reply README.md:39]: See the [YouTube video tutorial](https://www.youtube.com/watch?v=BaEJ8v7INNQ&list=PLN7ajX_VdyaNll_Aw3M-NIQ042BV5teP9&index=6) or the video on [user documentation migration](https://www.youtube.com/watch?v=KB-NPlRvLoY&list=PLN7ajX_VdyaNll_Aw3M-NIQ042BV5teP9&index=7).

Comment (eshack94) [reply README.md]: Thanks I think I figured it out. Added to `docs/images/` directory and then linked from there within the README file. Thanks. I think I'm ready for re-review.

**Comment by MarkEWaite** [comment]
My understanding was that the `hacktoberfest-accepted` label is only needed for pull requests that are not merged but should still be accepted as a valid Hacktoberfest contribution.  If the pull request is merged, then as far as I understand it, no specific label is required on the pull request, so long as the repository has been tagged.

**Comment by halkeye** [comment]
> My understanding was that the `hacktoberfest-accepted` label is only needed for pull requests that are not merged but should still be accepted as a valid Hacktoberfest contribution. If the pull request is merged, then as far as I understand it, no specific label is required on the pull request, so long as the repository has been tagged.

or the repo needs the topic, which this repo doesn't have

**Comment by eshack94** [comment]
@MarkEWaite I think you're right, however this repo isn't tagged with the `hacktoberfest` topic. I think it's either/or (i.e., `hacktoberfest` required on the repo then all merges count, or `hacktoberfest-accepted` required on individual PRs without the `hacktoberfest` topic present). It's a bit confusing for sure.

**Review by halkeye — APPROVED** [review]
LGTM

I can't merge or create labels in this repo though, so you'll have to wait for one of the admins

File: README.md [hunk 4]
```diff
@@ -1,6 +1,193 @@
...
+        (e.g. allow running only "QA\_.\*" jobs on a node)
+    -   Example 2: Prevent execution of user jobs on the Jenkins master
+        node
+-   Restrictions of jobs triggering by external causes
+    -   Example 1: Prohibit manual builds
+    -   Example 2: Allow triggering only by a job with the
+        specified [owner](https://wiki.jenkins.io/display/JENKINS/Ownership+Plugin).
+-   Several built-in restrictions + JobRestriction extension point
+
+### Contributors:
```
Comment (oleg-nenashev) [inline README.md:24]:
```suggestion
## Acknowledgements
```

File: README.md [hunk 5]
```diff
@@ -1,6 +1,193 @@
...
+
+1.  [Oleg Nenashev](hhttp://www.linkedin.com/in/onenashev) (maintainer)
+    -   Initial plugin version has been sponsored by Synopsys Inc.,
+        [www.synopsys.com](http://www.synopsys.com/)
```
Comment (oleg-nenashev) [inline README.md:28]:
```suggestion
Initial plugin version has been sponsored by [Synopsys, Inc.](http://www.synopsys.com/)
```

No need to mention me anymore

File: README.md [hunk 6]
```diff
@@ -1,6 +1,193 @@
 Job Restrictions Plugin for [Jenkins CI][1]
```
Comment (oleg-nenashev) [inline README.md:1]:
```suggestion
Job Restrictions Plugin for Jenkins
```

File: README.md [hunk 7]
```diff
@@ -1,6 +1,193 @@
...
+    installations on several platforms.  
+    Use job-restrictions 0.2.2 to avoid this issue
+
+## Version 0.1 (08/26/2013)
+
+-   ![(plus)](docs/images/add.svg)
+    Initial version with basic restrictions for nodes and jobs
+
 
 [1]: http://jenkins-ci.org/
```
Comment (oleg-nenashev) [inline README.md:192]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -1,6 +1,193 @@
 Job Restrictions Plugin for [Jenkins CI][1]
 =======================
-See [Job Restrictions Plugin Page][2] for more information.
+
+The plugin allows restricting job executions in order to change their
+behavior or to harden the security. With this plugin it is possible to
+configure nodes to accept only particular jobs. It is also possible to
+prevent job triggering by users and other jobs using various conditions.
+
+# About the plugin
```
Comment (oleg-nenashev) [inline README.md:9]:
```suggestion
```

The paragraph above represents the About section. The header is not needed IMHO

File: README.md [hunk 9]
```diff
@@ -1,6 +1,193 @@
...
+
+### Features:
+
+-   Execution restrictions on the node level
+    -   Example 1: Take jobs according to the specified name pattern
+        (e.g. allow running only "QA\_.\*" jobs on a node)
+    -   Example 2: Prevent execution of user jobs on the Jenkins master
```
Comment (oleg-nenashev) [inline README.md:16]:
```suggestion
    -   Example 2: Prevent execution of user jobs on the Jenkins controller
```

The terminology has been changed recently

File: README.md [hunk 10]
```diff
@@ -1,6 +1,193 @@
...
-See [Job Restrictions Plugin Page][2] for more information.
+
+The plugin allows restricting job executions in order to change their
+behavior or to harden the security. With this plugin it is possible to
+configure nodes to accept only particular jobs. It is also possible to
+prevent job triggering by users and other jobs using various conditions.
+
+# About the plugin
+
+### Features:
```
Comment (oleg-nenashev) [inline README.md:11]:
```suggestion
## Features
```

File: README.md [hunk 11]
```diff
@@ -1,6 +1,193 @@
...
+        specified [owner](https://wiki.jenkins.io/display/JENKINS/Ownership+Plugin).
+-   Several built-in restrictions + JobRestriction extension point
+
+### Contributors:
+
+1.  [Oleg Nenashev](hhttp://www.linkedin.com/in/onenashev) (maintainer)
+    -   Initial plugin version has been sponsored by Synopsys Inc.,
+        [www.synopsys.com](http://www.synopsys.com/)
+
+# Usage
```
Comment (oleg-nenashev) [inline README.md:30]:
```suggestion
## Usage
```

File: README.md [hunk 12]
```diff
@@ -1,6 +1,193 @@
...
+
+## Job restrictions for nodes
```
Comment (oleg-nenashev) [inline README.md:32]:
```suggestion
### Job restrictions for nodes
```

File: README.md [hunk 13]
```diff
@@ -1,6 +1,193 @@
...
+
+These restrictions can be configured in node's configuration page.
+
+-   Node won't accept jobs, which violate the specified restrictions
+-   Other requirements (executors, labels, etc.) will be checked as well
+
+![](docs/images/Jenkins_JobRestrictions_NodeRestriction.png)
+
+## Job restrictions for projects
```
Comment (oleg-nenashev) [inline README.md:41]:
```suggestion
### Job restrictions for projects
```

File: README.md [hunk 14]
```diff
@@ -1,6 +1,193 @@
...
+[JENKINS-19497](https://issues.jenkins-ci.org/browse/JENKINS-19497)).
+
+The current version supports the following checks:
+
+-   **Apply Job restrictions to upstream build** - prohibits the
+    invocation from specific builds, which do not satisfy the specified
+    requirements
+-   Prohibit manual launch of the job (since 0.2)
+
+# Extension points
```
Comment (oleg-nenashev) [inline README.md:56]:
```suggestion
## Extension points
```

File: README.md [hunk 15]
```diff
@@ -1,6 +1,193 @@
...
+    -   [Rebuild
+        Plugin](https://wiki.jenkins.io/display/JENKINS/Rebuild+Plugin)
+        and other specific causes are not supported
+-   Regex restriction - check the jobs name by a regular expression
+
+The following plugins produce additional job restrictions:
+
+-   Page:
+    [Ownership
+    Plugin](https://wiki.jenkins.io/display/JENKINS/Ownership+Plugin)
```
Comment (oleg-nenashev) [inline README.md:75]:
```suggestion
-  [Ownership Plugin](http://plugins.jenkins.io/ownership)
```

File: README.md [hunk 16]
```diff
@@ -1,6 +1,193 @@
...
+
+-   Page:
+    [Priority Sorter
+    Plugin](https://wiki.jenkins.io/display/JENKINS/Priority+Sorter+Plugin)
```
Comment (oleg-nenashev) [inline README.md:79]:
```suggestion
-  [Priority Sorter Plugin](https://plugins.jenkins.io/PrioritySorter/)
```

File: README.md [hunk 17]
```diff
@@ -1,6 +1,193 @@
...
+If your plug-in is not listed here, then simply add the label
+**job-restriction-producer** to your plug-in wiki page and it will be
+automatically listed.
+
+# JIRA issues
+
+If you have any proposals/bug reports, please create an issue on [Jenkins
+JIRA][2].
+
+# Version history
```
Comment (oleg-nenashev) [inline README.md:92]: I would recommend moving it to a separate CHANGELOG.md file

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Thanks for the pull request! There are some issues in the pull request, mostly header level. Happy to merge once the issues are fixed

File: README.md (hunk 4 above)
Comment (eshack94) [reply README.md]: Change made! Thanks for the feedback.

File: README.md (hunk 5 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 6 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 7 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 8 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 9 above)
Comment (eshack94) [reply README.md]: Change made!

File: README.md (hunk 10 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 11 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 12 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 13 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 14 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 15 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 16 above)
Comment (eshack94) [reply README.md]: Change made.

File: README.md (hunk 17 above)
Comment (eshack94) [reply README.md]: Moved.

**Review by oleg-nenashev — APPROVED** [review]
Thanks! I will go ahead and merge it. More refactoring can be done later

**Pull request merged**